File: python/recursive_linear_search.py
def recursive_linear_search(A,n,i,x):
  if i>=n:
    return "NOT FOUND"
  elif A[i]==x:
    return i
  else:
    return recursive_linear_search(A,n,i+1,x)
    
A=[1,2,3,4,5]
print(recursive_linear_search(A,len(A),0,6))

File: javascript/recursive_linear_search.js
function recursive_linear_search(A,n,i,x) {
	if(i>=n){
  	return "NOT FOUND"
  }
  else{
  	if(A[i]==x){
    	return i;
    }
    else{
    	return recursive_linear_search(A,n,i+1,x);
    }
  }
}


var a = recursive_linear_search([1,2,3],[1,2,3].length,0,5);
alert(a)